### PR TITLE
Adding the --build-config option

### PR DESCRIPTION
**Pull request by ematejska**
Adding the --build-config option which gives the ability to either build with the debug or release configuration.   Capitalization is adjusted to accommodate a Swift package or an Xcode project as an implementation detail.

### CODE REVIEW

**Comment by ematejska** [comment]
@swift-ci please test source compatibility

**Comment by ematejska** [comment]
@swift-ci test source compatibility

**Comment by ematejska** [comment]
@swift-ci test

File: project.py [hunk 1]
```diff
@@ -443,8 +449,14 @@ def add_arguments(parser):
...
-                        action='store_true')
-
+                        action='store_true'),
+    parser.add_argument("--build-config",
+                        metavar="NAME",
+                        choices=['debug', 'release'],
+                        dest='build_config',
+                        help='specify "debug" or "release" to override '
+                             'the build configuration in the projects.json file',
+                        default='')
```
Comment (shahmishal) [inline project.py]: We should set the default value to one that can be picked. (debug or release)

Comment (lplarson) [inline project.py]: When a config isn't specified, it shouldn't override the config specified in projects.json. We should just remove the default parameter and let it default to None. There's no need to manually specify an empty string here. I'll make the change and accept this pull request.

**Pull request merged**